### PR TITLE
Improve step pattern handling and argument parsing

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- cache compiled regex per step pattern and support typed placeholders and escaped braces
- validate duplicate argument names and enrich parse error messages
- test overflow behaviour for u32 step arguments

## Testing
- `make fmt`
- `make lint`
- `make test`
- `make markdownlint`

------
https://chatgpt.com/codex/tasks/task_e_688db84b37ac8322b078ae04311d9dd2

## Summary by Sourcery

Improve step pattern handling and argument parsing by introducing typed placeholders and escaped brace support, caching compiled regexes, unifying macro argument logic, validating duplicate names, enhancing error messages, optimizing step registration and lookup, and adding tests for argument overflow

New Features:
- Support typed placeholders in step patterns with type-based regex subpatterns
- Allow escaped braces (‘{{’, ‘}}’) in step patterns

Enhancements:
- Cache compiled regular expressions per StepPattern to avoid recompilation
- Unify fixture and step argument declaration logic in macros
- Enhance parse error messages to include placeholder name, raw value, and target type
- Validate duplicate argument names between fixtures and step parameters
- Switch to static StepPattern registration and new submit! macro for steps
- Optimize lookup_step with direct map lookup by pattern string

Documentation:
- Update design documentation to reflect regex caching, typed placeholders, and escaped brace support

Tests:
- Add test for u32 overflow in step argument parsing and invalid scenario error

### CODE REVIEW

File: crates/rstest-bdd/src/lib.rs [hunk 1]
```diff
@@ -203,24 +240,68 @@ pub fn extract_placeholders(pattern: PatternStr<'_>, text: StepText<'_>) -> Opti
...
+            }
+            '}' => {
+                if chars.peek() == Some(&'}') {
+                    chars.next();
+                    regex_source.push_str("\\}");
+                } else {
+                    regex_source.push_str("\\}");
+                }
+            }
+            _ => regex_source.push_str(&regex::escape(&ch.to_string())),
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/rstest-bdd/src/lib.rs:275]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.
